simulator/virtual_machine.go

```diff
@@ -2237,3 +2239,32 @@ func allSnapshotsInTree(tree []types.VirtualMachineSnapshotTree) []types.Managed
 
 	return result
 }
+
+func changeTrackingSupported(spec *types.VirtualMachineConfigSpec) *bool {
+	supported := false
+	for _, device := range spec.DeviceChange {
+		if dev, ok := device.GetVirtualDeviceConfigSpec().Device.(*types.VirtualDisk); ok {
+
+			if _, ok := dev.Backing.(*types.VirtualDiskFlatVer2BackingInfo); ok {
```

Comment:
Thanks @amanpaha , your first commit used a `switch dev.Backing.(type)`, which is what I would suggest here.  I think `changeTrackingSupported` can return `bool` instead of `*bool`, then each case can `return true` and the default case `return false`.  Then use the `NewBool` helper to convert to a pointer:
```go
vm.Capability.ChangeTrackingSupported = types.NewBool(changeTrackingSupported(spec))
```
What do you think?

Reply:
Yes, definitely looks good, thanks for the suggestion.
I have made the change, should I resolve the conversation or reviewer will do it after verifying that the correct changes were made.